src/components/learner-credit-management/cards/AssignmentModalSummary.jsx

```diff
@@ -1,44 +1,66 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Card, Stack } from '@edx/paragon';
+import classNames from 'classnames';
+import { Card, Stack, Icon } from '@edx/paragon';
+import { Error } from '@edx/paragon/icons';
 
-import { formatPrice } from '../data';
+import { formatPrice, useBudgetId, useSubsidyAccessPolicy } from '../data';
 import AssignmentModalSummaryEmptyState from './AssignmentModalSummaryEmptyState';
 import AssignmentModalSummaryLearnerList from './AssignmentModalSummaryLearnerList';
+import AssignmentModalSummaryErrorState from './AssignmentModalSummaryErrorState';
 
 const AssignmentModalSummary = ({
   course,
   learnerEmails,
+  inputValidationErrorMessage,
 }) => {
+  const { subsidyAccessPolicyId } = useBudgetId();
+  const { data: subsidyAccessPolicy } = useSubsidyAccessPolicy(subsidyAccessPolicyId);
+
   const learnerEmailsCount = learnerEmails.length;
   const hasLearnerEmails = learnerEmailsCount > 0;
+  const hasInputValidationError = hasLearnerEmails && !!inputValidationErrorMessage;
+  const hasValidLearnerEmails = hasLearnerEmails && !hasInputValidationError;
```

Comment:
[clarification] I am almost wondering if its worth passing the `isValid` value from the parent component (AssignmentModalContent) as a prop called `hasValidLearnerEmails`, since this logic is technically already resolved higher up in the component tree with function `isEmailAddressesInputValueValid`.
Ex.
```jsx
const AssignmentModalContent = ({ course, onEmailAddressesChange }) => {
    const [hasValidLearnerEmails, setHasValidLearnerEmails] = useState(false);
    if (isValid) {
      setInputValidationErrorMessage(null);
      setHasValidLearnerEmails(isValid && learnerEmails.length > 0);
      onEmailAddressesChange(emails);
    } else {
    etc...
    
return ( 
            <AssignmentModalSummary
              course={course}
              learnerEmails={learnerEmails}
              inputValidationErrorMessage={inputValidationErrorMessage}
              hasValidLearnerEmails = {hasValidLearnerEmails}
              hasEnoughBalanceForAssignments
            />
)
}
```

Reply:
Yep, totally agree! Good catch. This was one of the remaining areas of clean up for sure.

Reply:
@brobro10000 I've updated the approach slightly. Now, whenever the `learnerEmails` change within `AssignmentModalContent`, the validation will be run through `isEmailAddressesInputValueValid` and pass its return value down to `AssignmentModalSummary` to be more DRY.